src/main/java/com/labsynch/labseer/service/SaltServiceImpl.java

```diff
@@ -99,4 +169,369 @@ private void saveSalt(CmpdRegMolecule mol) throws IOException, CmpdRegMolFormatE
+
+					if(Parent.countParents() > 0)
+					{
+						// findParentsBySaltForms(Set<SaltForm> saltForms)
+						TypedQuery<Parent> parentQuery = Parent.findParentsByAnySaltForms(saltFormsSet);
+						parents = parentQuery.getResultList();
+					}
+				}
+				
+			}
+			// Returns a List of All Parents 
+
+			for (Parent parent : parents)
+			{
+				parentIDs.add(parent.getId());
+			}
+	
+			return parentIDs;
+		}
+		catch (NullPointerException e)
+		{
+			// Query Result Empty
+			e.printStackTrace();
+			return new ArrayList<Long>();
+		}
+	}
+
+	// Helper Method Used to Check Dependencies Within ACAS
+	private Map<String, HashSet<String>> checkACASDependencies(
+		Map<String, HashSet<String>> acasDependencies) throws MalformedURLException, IOException {
+			String url = propertiesUtilService.getAcasURL() + "compounds/checkBatchDependencies";
+			BatchCodeDependencyDTO request = new BatchCodeDependencyDTO(acasDependencies.keySet());
+			String json = request.toJson();
+			String responseJson = SimpleUtil.postRequestToExternalServer(url, json, null);
+			BatchCodeDependencyDTO responseDTO = BatchCodeDependencyDTO.fromJsonToBatchCodeDependencyDTO(responseJson);
+			if (responseDTO.getLinkedDataExists()) {
+				for (CodeTableDTO experimentCodeTable : responseDTO.getLinkedExperiments()) {
+					String experimentCodeAndName = experimentCodeTable.getCode() + " ( " + experimentCodeTable.getName()
+							+ " )";
+					if (acasDependencies.containsKey(experimentCodeTable.getComments())) {
+						acasDependencies.get(experimentCodeTable.getComments()).add(experimentCodeAndName);
+					} else {
+						HashSet<String> codes = new HashSet<String>();
+						codes.add(experimentCodeAndName);
+						acasDependencies.put(experimentCodeTable.getComments(), codes);
+					}
+				}
+			}
+			return acasDependencies;
+		}
+
+	// Helper Method to Check Dependent Containers; Utilized in Method Below It
+
+	// batchCodes == Lot Corp Name
+	private Collection<ContainerBatchCodeDTO> checkDependentACASContainers(Set<String> batchCodes)
+		throws MalformedURLException, IOException {
+			String url = propertiesUtilService.getAcasURL() + "containers/getContainerDTOsByBatchCodes";
+			String json = (new JSONSerializer()).serialize(batchCodes);
+			String responseJson = SimpleUtil.postRequestToExternalServer(url, json, null);
+			Collection<ContainerBatchCodeDTO> responseDTOs = ContainerBatchCodeDTO
+					.fromJsonArrayToContainerBatchCoes(responseJson);
+			return responseDTOs;
+		}
+
+	// Method to Check for Dependent Data 
+	public PurgeSaltDependencyCheckResponseDTO checkDependentData(Salt salt)
+	{
+		// Get IsoSalts From Salt 
+		TypedQuery<IsoSalt> isoSaltQuery = IsoSalt.findIsoSaltsBySalt(salt);
+		Collection<IsoSalt> isoSalts =  isoSaltQuery.getResultList(); 
+		Set<IsoSalt> isoSaltsSet = new HashSet<IsoSalt>(isoSalts);
+
+		// Getter of IsoSalt to SaltForm 
+
+		Set<SaltForm> saltFormSet = new HashSet<SaltForm>();
```

Comment:
same nit here about combining the for loops

Reply:
nit: If you really wanted to DRY this up, you might consider a small helper function: `findLotsBySalt` that takes this bit of repeated logic out of both `checkDependentData` and `updateDependencies`